Backend/app/provider-platform/dynamic-offer-driver-app/Main/src/Domain/Action/Dashboard/Ride.hs

```diff
@@ -244,51 +244,65 @@ mkBookingStatus ride now = do
     DRide.CANCELLED -> Common.CANCELLED
 
 ---------------------------------------------------------------------
-rideSync :: ShortId DM.Merchant -> Id Common.Ride -> Flow Common.RideSyncRes
-rideSync merchantShortId reqRideId = do
+rideSync :: ShortId DM.Merchant -> Common.RideSyncReq -> Flow Common.RideSyncRes
+rideSync merchantShortId rideSyncReq = do
   merchant <- findMerchantByShortId merchantShortId
-  let rideId = cast @Common.Ride @DRide.Ride reqRideId
-  ride <- runInReplica $ QRide.findById rideId >>= fromMaybeM (RideDoesNotExist rideId.getId)
-  booking <- runInReplica $ QBooking.findById ride.bookingId >>= fromMaybeM (BookingNotFound rideId.getId)
+  let rideIds = map (cast @Common.Ride @DRide.Ride) rideSyncReq.rideIds
+  rides <- mapM (\rideId -> runInReplica $ QRide.findById rideId >>= fromMaybeM (RideDoesNotExist rideId.getId)) rideIds
+  let rideIds_rides_zip = zip rideIds rides
```

Comment:
Why zip? We can get `id` from `ride` itself

Reply:
changed it

Reply:
Also please use `camelCase` in Haskell code

Reply:
change it